modules/core/src/smithy4s/schema/Alt.scala

```diff
@@ -26,7 +27,8 @@ import kinds._
 final case class Alt[U, A](
     label: String,
     schema: Schema[A],
-    inject: A => U
+    inject: A => U,
+    project: PartialFunction[U, A]
```

Comment:
New projection function, which removes the need to have `projector` in `Dispatcher`

Reply:
I thought about it and I don't really want to because it'd impact things too much. Moreover we're making the generation of optics an opt-in.

Reply:
And we can't make the `Alt` a bonafide `Prism` and expose it to users because of recursion concerns. Alts need to be hidden in the computation of the schema.

Reply:
hmm I don't know about the recursion concerns, can you elaborate?

other than that I get it

Reply:
> hmm I don't know about the recursion concerns, can you elaborate?

Actually I was partially wrong. What I meant was that : 

```scala 
val schema: Schema[A] = Schema.recursive {
    val aField = schema.optional[A]("a", _.a) 
    Schema.struct(aField)(A(_))
} 
```

Rendering it as follows to make `aField` visible to the external world doesn't work because `aField` refers to the initialised schema. 

```scala 
val aField = schema.optional[A]("a", _.a) 
val schema: Schema[A] = Schema.recursive {
    Schema.struct(aField)(A(_))
} 
```

However we could potentially render it as follows, which works : 

```scala 
val schema: Schema[A] = Schema.recursive {
    Schema.struct(aField)(A(_))
} 

val aField = schema.optional[A]("a", _.a) 
```